src/plotting.jl

```diff
@@ -30,81 +30,95 @@ end
 """
 Plot prediction of the GP as a soft decision boundary between [0,1].
 """
-function plot_soft_boundary(gp, models; m=[200,200], show_data=true, overlay=false, overlay_levels=50, ms=4, lw=0, tight=false) # lw=0.2
-    y = gp_output(gp, models; m)
-    model_ranges = get_model_ranges(models, size(y))
+function plot_soft_boundary(gp, models; inds=[:, :, fill(1, length(models) - 2)...], num_steps=200, m=fill(num_steps, length(models)), show_data=true, overlay=false, overlay_levels=50, ms=4, lw=0, tight=false) # lw=0.2
+    y = gp_output(gp, models; m)[inds...]
+    full_models = [model for (model, d) in zip(models, inds) if d == :]
+    model_ranges = get_model_ranges(full_models, size(y))
     contourf(model_ranges[1], model_ranges[2], y'; c=COLOR_FAIL, lc=:black, lw=lw, clims=(0,1))
     if overlay
-        p(x,y) = pdf(models[1].distribution, x) * pdf(models[2].distribution, y)
+        p(x,y) = pdf(full_models[1].distribution, x) * pdf(full_models[2].distribution, y)
         @suppress contour!(model_ranges[1], model_ranges[2], p, c=cgrad([:gray, :black, :black, :white], 10, categorical=true, scale=:exp, rev=false), levels=overlay_levels)
     end
     if show_data
-        plot_data!(gp.x, gp.y; ms)
+        inds = [i for (i, d) in enumerate(inds) if d == :]
```

Comment:
Could this be obtained at the same time we create `full_models` above instead of going through the loop again?

Reply:
rewrote this a bit so we find the indices of the sliced dimensions once at the top